packages/gatsby/src/bootstrap/load-plugins/load.js

```diff
@@ -211,13 +211,29 @@ module.exports = (config = {}, rootDir = null) => {
     )
   })
   const program = store.getState().program
+
+  // default options for gatsby-plugin-page-creator
+  let pageCreatorOptions = {
+    path: slash(path.join(program.directory, `src/pages`)),
+    pathCheck: false,
+  }
+
+  if (config.plugins) {
+    const pageCreatorPlugin = config.plugins.find(
+      plugin =>
+        plugin.resolve === `gatsby-plugin-page-creator` &&
+        plugin.options.path.match(/src\/pages$/)
```

Comment:
It should not match only the last chars but the full path, because all path for this plugin are full path

Reply:
can be also `${__dirname}/myguides/src/pages`

Reply:
Updated the condition to override the defaults only if the path matches <gatsby-site-root>/src/pages